### PR TITLE
Added route for 'shop all' page

### LINKED ISSUE #83: (SP: ) Create page route for products
When user press 
image 
button should to include route for `all products` like `/products`

Should create after #77

### PR DESCRIPTION
**Pull request by heshvab**
- Added page file
- Added route for it inside of routes
- Added `BrowserRouter` in categories' tests because I added `Link` inside of `Categories` component. Would be good to change it with `renderWithProviders` in #63

### CODE REVIEW

File: code/src/routes/routes.tsx [hunk 1]
```diff
@@ -17,6 +18,16 @@ const routes: RouteObject[] = [
...
+  {
+    path: "/products",
+    errorElement: <ErrorPage />,
+    children: [
+      {
+        element: <RootLayout />,
+        children: [{ index: true, element: <ProductsPage /> }]
+      }
+    ]
+  },
```
Comment (mhevyk) [inline code/src/routes/routes.tsx]: it is not correct. Our routes should look like following:

```jsx
{
    path: "/",
    errorElement: <ErrorPage />,
    children: [
      {
        element: <RootLayout />,
        children: [
           { index: true, element: <HomePage /> },
           { path: "products", element: <ProductsPage /> }
        ]
      }
    ]
  },
```

**Review by OliaHrytsak — APPROVED** [review]

File: code/src/routes/routes.tsx (hunk 1 above)
Comment (D1nviz) [inline code/src/routes/routes.tsx]: I suggest creating object with routes and exporting it as const. For example
```
const routes = {
  home: "/",
  products: "/products",
  notFound ....
 } as const 
```

Comment (heorhiiss) [reply code/src/routes/routes.tsx]: Thanks for suggestion. I created routes file in `constants`. It has similar to s2s structure with `path` and `route` properties.

**Review by mhevyk — CHANGES_REQUESTED** [review]

File: code/src/layouts/app-header/components/categories/Categories.test.tsx [hunk 2]
```diff
@@ -1,10 +1,15 @@
 import { render, screen } from "@testing-library/react";
+import { BrowserRouter } from "react-router-dom";
 import Categories from "@/layouts/app-header/components/categories/Categories";
 import { categories } from "@/layouts/app-header/components/categories/Categories.constants";
 
 describe("Categories Component", () => {
```
Comment (mhevyk) [inline code/src/layouts/app-header/components/categories/Categories.test.tsx]: Why component in uppercase?

File: code/src/layouts/app-header/components/categories/Categories.tsx [hunk 3]
```diff
@@ -1,3 +1,5 @@
+import { Link } from "react-router-dom";
```
Comment (mhevyk) [inline code/src/layouts/app-header/components/categories/Categories.tsx]: Use `AppLink` instead

**Review by mhevyk — CHANGES_REQUESTED** [review]

File: code/src/layouts/app-header/components/categories/Categories.test.tsx (hunk 2 above)
Comment (heorhiiss) [reply code/src/layouts/app-header/components/categories/Categories.test.tsx]: It was not written by me. I fixed it.

File: code/src/layouts/app-header/components/categories/Categories.tsx (hunk 3 above)
Comment (heorhiiss) [reply code/src/layouts/app-header/components/categories/Categories.tsx]: Thanks, used AppLink there.

File: code/src/routes/routes.tsx (hunk 1 above)
Comment (heorhiiss) [reply code/src/routes/routes.tsx]: Thank you, fixed it according to provided example.

**Review by mhevyk — APPROVED** [review]

File: code/src/constants/routes.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+const routes = {
+  home: {
+    path: "/"
+  },
+  products: {
+    path: "/products",
+    route: "products"
```
Comment (D1nviz) [inline code/src/constants/routes.ts]: Why you are using route property? Do we really need it?

Comment (heorhiiss) [reply code/src/constants/routes.ts]: Well, in this case both properties can be used in the same way, so `route` can be removed. But in some cases, we need absolute path (`path` property) as well as relative path (relative to parent's path, `route` property). `Path` we can use in `Link` for example. `route` property we can use inside of our router, where each route can accept children whose paths must be relative to their parent's path. Maybe this needs better naming, cause just `route` and `path` in this case are not very descriptive.

**Review by D1nviz — APPROVED** [review]

**Review by dmtrth25 — APPROVED** [review]

**Pull request merged**